src/hooks/tradingView/useChartLines.ts

```diff
@@ -189,59 +222,187 @@ export const useChartLines = ({
     });
   }, [stringGetter, currentMarketId, currentMarketPositionData, maybeDrawPositionLine]);
 
+  // Cache for order modification that stores the new orders that are submitted but not yet placed
+  const pendingOrderAdjustmentsRef = useRef<{
+    [clientId: string]: { orderPayload: HumanReadablePlaceOrderPayload; oldOrderId: string };
+  }>({});
+
+  const removePendingOrderAdjustment = (clientId: number) => {
+    const { [clientId]: removed, ...withoutOrderId } = pendingOrderAdjustmentsRef.current;
+    pendingOrderAdjustmentsRef.current = withoutOrderId;
+  };
+
+  const addPendingOrderAdjustment = (
+    orderPayload: HumanReadablePlaceOrderPayload,
+    oldOrderId: string
+  ) => {
+    pendingOrderAdjustmentsRef.current = {
+      ...pendingOrderAdjustmentsRef.current,
+      [orderPayload.clientId]: { orderPayload, oldOrderId },
+    };
+  };
+
+  const dispatch = useAppDispatch();
+
+  const onMoveOrderLine = useCallback(
+    async (order: SubaccountOrder, orderLine?: IOrderLineAdapter) => {
+      if (!orderLine || !canModifyOrderTypeFromChart(order)) return;
+
+      const oldPrice = order.triggerPrice ?? order.price;
+      const newPrice = orderLine.getPrice();
+
+      // TODO(tinaszheng): do validation here for new price
+      // make sure the newPrice doesnt cross over the current price depending
+      // on the direction of the trade
+
+      orderLine.setPrice(newPrice);
+      // Don't go through abacus for limit order modifications to avoid having to override any trade inputs in the Trade Form
+      const orderPayload = createPlaceOrderPayloadFromExistingOrder(order, newPrice);
+      if (!orderPayload) {
+        orderLine.setPrice(oldPrice);
+        return;
+      }
```

Comment:
nit: should we just do the following, or is it because we'll show an error?
```
const orderPayload = createPlaceOrderPayloadFromExistingOrder(order, newPrice);
if (!orderPayload) return;
orderLine.setPrice(newPrice);
```

Reply:
oh nice! updated